### DISCUSSION TITLE
PAC file?

### DISCUSSION THREAD

**Discussion by maxim-zodchy** [xvzc/SpoofDPI#157] [Ideas]
Hi! Is it possible to make ability to use PAC file for split routing? Everybody in Russia need to go directly to gosuslugi.ru, etc. It would be very useful  if we can add some *.TLD URLs in some list in a file and SpoofDPI does not make it's work with theese URL's. Thank you in advance!

**Comment by asimba** [comment]
`
function FindProxyForURL(url, host){
  if (dnsDomainIs(host, ".googlevideo.com")) {
    return "PROXY 127.0.0.1:8080";
  }
  if (dnsDomainIs(host, ".youtube.com")) {
    return "PROXY 127.0.0.1:8080";
  }
  if (dnsDomainIs(host, ".ytimg.com")) {
    return "PROXY 127.0.0.1:8080";
  }
  if (dnsDomainIs(host, ".ggpht.com")) {
    return "PROXY 127.0.0.1:8080";
  }
  if (dnsDomainIs(host, "play.google.com")) {
    return "PROXY 127.0.0.1:8080";
  }
  if (dnsDomainIs(host, ".nhacmp3youtube.com")) {
    return "PROXY 127.0.0.1:8080";
  }
  if (dnsDomainIs(host, ".wixmp.com")) {
    return "PROXY 127.0.0.1:8080";
  }
  return "DIRECT";
}
'`
Вполне себе работает. В системе он как обычный прокси висит (в отличие от goodbyedpi).

**Reply by maxim-zodchy** [reply]
Спасибо, а эту функцию куда следует прикрутить? Создать из этого кода (изменив требуемые URL) PAC-файл и указать системе на него, я правильно понял? Тогда вопрос – в коде указано, какие ресурсы подлежат проксированию, задача изначально обратная: исключить определенные URL, прописав только их, а остальные по-умолчанию проксировать. PS: под MacOS внезапно © обнаружил в настройках прокси форму: "Не использовать прокси для следующих узлов и доменов". При семантике "*domain1.tld, *domain2.tld" решение рабочее, "Госуслуги" открылись.

**Reply by maxim-zodchy** [reply]
Пардон за оффтоп: а разве GoodbyeDPI (кстати, у него нет проблем с открытием вышеуказанного ресурса "из коробки") – не HTTP-прокси? Я видел там тот же адрес в настройках с другим портом: 127.0.0.1:10812. В чем подвох?

**Reply by asimba** [reply]
1) Да, это код рабочего proxy.pac. Для конкретного домена можно сделать `return "DIRECT";` и `return "PROXY 127.0.0.1:8080";` по-умолчанию. Т.е. сделать наоборот... вроде:
`
function FindProxyForURL(url, host){
  if (dnsDomainIs(host, ".gosuslugi.ru")) {
    return "DIRECT";
  }
  return "PROXY 127.0.0.1:8080";
}
`
2) Оффтоп не прокси, а фильтр - он через драйвер WinDivert вклинивается и модифицирует проходящие пакеты (или все или по списку доменов). Во всяком случае, я так понял из описания. Там вообще прокси можно не указывать.
